tst/unit/test_swarm.cpp

```diff
@@ -212,8 +212,8 @@ TEST_CASE("Swarm memory management", "[Swarm]") {
 
   // Check that data was moved during defrag
   x_h = swarm->Get<Real>(swarm_position::x::name()).Get().GetHostMirrorAndCopy();
-  REQUIRE(x_h(2) == 1.2);
-  REQUIRE(x_h(4) == 1.1);
+  REQUIRE(x_h(2) == 1.1);
+  REQUIRE(x_h(4) == 1.2);
```

Comment:
This change is because the previous defragmentation operation was flipping the ordering of particles that got moved, whereas the new method preserves the ordering during the move.